test/ng/httpSpec.js

```diff
@@ -1369,17 +1369,15 @@ describe('$http', function() {
             }
           );
 
-          it('should forward json deserialization errors to the http error handler',
-              function() {
+          it('should return JSON data with error message if JSON is invalid', function() {
             var errCallback = jasmine.createSpy('error');
-
-            $httpBackend.expect('GET', '/url').respond('abcd', {'Content-Type': 'application/json'});
-            $http({method: 'GET', url: '/url'}).then(callback).catch(errCallback);
+            $httpBackend.expect('GET', '/url').respond('{abcd}', {'Content-Type': 'application/json'});
```

Comment:
I've replaced the `it` function of `forwarding json deserialization errors` with mine. The above highlighted statements are pretty much similar to the old code except that we're using an object `'{abcd}'` within the string instead of a direct string `'abcd'`. Are you referring to that string or something else?